packages/mui-base/src/Select/Select.test.tsx

```diff
@@ -19,12 +21,28 @@ import { OptionGroup } from '@mui/base/OptionGroup';
 // currently the setup() method uses the ClipboardEvent constructor which is incompatible with our lowest supported version of iOS Safari (12.2) https://github.com/mui/material-ui/blob/master/.browserslistrc#L44
 // userEvent.setup() requires Safari 14 or up to work
 
+async function flushMicrotasks() {
+  if (/jsdom/.test(window.navigator.userAgent)) {
+    // This is only needed for JSDOM and causes issues in real browsers
+    await act(() => async () => {});
+  }
+}
+
 describe('<Select />', () => {
   const mount = createMount();
-  const { render } = createRenderer();
+  const { render: internalRender } = createRenderer();
+
+  async function render(
```

Comment:
nit: `render` might be a confusing name as most tests use `render` returned by `createRenderer` directly. I would name this `renderAndFlush`